src/hotspot/share/gc/g1/heapRegion.hpp

```diff
@@ -171,7 +171,7 @@ class HeapRegion : public CHeapObj<mtGC> {
   // Update heap region that has been compacted to be consistent after Full GC.
   void reset_compacted_after_full_gc();
   // Update pinned heap region (not compacted) to be consistent after Full GC.
-  void reset_pinned_after_full_gc();
+  void reset_not_compacted_after_full_gc();
```

Comment:
Now that this version uses the existing "pinned" mechanism to skip high-live-ratio regions, this method can retain its original name/implementation, right?

Reply:
That is true, full gc reuses the mechanism originally implemented for pinned regions. However "pinned" is something very specific in G1 context so I think it is better to use a different, more generic name. The regions that are not compacted (and not yet pinned) are not really temporarily pinned.

There has been an earlier discussion (not sure if here in this PR) to actually rename the use of "pinned" in G1 full gc to use "not compacted" too, resulting in a CR to rename this (in [JDK-8264423](https://bugs.openjdk.java.net/browse/JDK-8264423)).

However I was going to wait for this change to go in before sending out a PR.

Also this "not_compacted" name better matches the "compacted" method name above.

So I would prefer to keep this as is.

Reply:
I see; agree.